src/db/collections/mod.rs

```diff
@@ -32,56 +32,34 @@ pub use self::{
 };
 use crate::types::stardust::block::{
     output::{AliasOutput, BasicOutput, FoundryOutput, NftOutput},
-    payload::{MilestonePayload, TaggedDataPayload, TransactionPayload, TreasuryTransactionPayload},
+    Output,
 };
 
 /// Helper to specify a kind for an output type.
-pub trait OutputKind {
+pub trait OutputKindQuery {
     /// Gets the output kind.
-    fn kind() -> Option<&'static str> {
-        None
-    }
+    fn kind() -> Option<&'static str>;
```

Comment:
what about making this:
```rust
const KIND: &'static str;
```
and let `Output` KIND be `""`?

Reply:
No that won't work with the query building.